packages/server/src/handlers/BaseHandler.ts

```diff
@@ -158,7 +162,81 @@ export class BaseHandler extends EventEmitter {
         }
       })
 
-      this.store.write(req.pipe(proxy), id, offset).then(resolve).catch(reject)
+      stream
+        .pipeline(req.pipe(proxy), new StreamLimiter(maxFileSize), async (stream) => {
+          return this.store.write(stream as StreamLimiter, id, offset)
+        })
+        .then(resolve)
+        .catch(reject)
     })
   }
+
+  getConfiguredMaxSize(req: http.IncomingMessage, id: string) {
+    if (typeof this.options.maxSize === 'function') {
+      return this.options.maxSize(req, id)
+    }
+    return this.options.maxSize ?? 0
+  }
+
+  /**
+   * Calculates the maximum allowed size for the body of an upload request.
+   * This function considers both the server's configured maximum size and
+   * the specifics of the upload, such as whether the size is deferred or fixed.
+   */
+  async calculateMaxBodySize(
```

Comment:
This logic in this function appears a bit confusing to me. I wasn't able to spot a mistake, but the number of conditionals makes it hard to reason about.

I don't know exactly how it could be improved but maybe it's possible to refactor this into one overarching condition checking whether the size is deferred or not.

```
if file.sizeIsDeferred {

} else {

}
```

And then have the additional checks in its branches.

Reply:
Cool! I agree!
I initially ported it as is from tusd, but I agree with your comment.

I have now cleaned it up - let me know if it looks good now